test/integration/framework/etcd.go

```diff
@@ -172,12 +172,27 @@ func EtcdMain(tests func() int) {
 	// Bail out early when -help was given as parameter.
 	flag.Parse()
 
+	before := runtime.NumGoroutine()
 	stop, err := startEtcd()
 	if err != nil {
 		klog.Fatalf("cannot run integration tests: unable to start etcd: %v", err)
 	}
 	result := tests()
 	stop() // Don't defer this. See os.Exit documentation.
+
+	// Log the number of goroutines leaked.
+	// TODO: we should work on reducing this as much as possible.
+	var dg int
+	for i := 0; i < 10; i++ {
+		if dg = runtime.NumGoroutine() - before; dg <= 4 {
```

Comment:
comment on why 4? seems magical

Reply:
I can imagine things like the flush routine in klog we'd never get rid of... a floor is fine, just comment why